src/Integration/TeamExplorer/SectionController.cs

```diff
@@ -0,0 +1,277 @@
+//-----------------------------------------------------------------------
+// <copyright file="SectionController.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.TeamFoundation.Client.CommandTarget;
+using Microsoft.TeamFoundation.Controls;
+using Microsoft.TeamFoundation.Controls.WPF.TeamExplorer;
+using SonarLint.VisualStudio.Integration.Progress;
+using SonarLint.VisualStudio.Integration.WPF;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Windows.Input;
+
+namespace SonarLint.VisualStudio.Integration.TeamExplorer
+{
+    /// <summary>
+    /// Controller for the SonarLint section in team explorer tool window
+    /// The class is responsible for view and view model creation also hosting the commands
+    /// relevant during the life time of the section (initialized when activated and disposed when navigated to a different section).
+    /// </summary>
+    [TeamExplorerSection(SectionController.SectionId, SonarQubePage.PageId, SectionController.Priority)]
+    internal class SectionController : TeamExplorerSectionBase, IConnectSection
+    {
+        public const string SectionId = "25AB05EF-8132-453E-A990-55587C0C5CD3";
+        public const int Priority = 300;
+
+        internal const int CommandNotHandled = (int)OleConstants.OLECMDERR_E_UNKNOWNGROUP;
+
+        private readonly IWebBrowser webBrowser;
+
+        [ImportingConstructor]
+        public SectionController([Import] IHost host, IWebBrowser webBrowser)
```

Comment:
This constructor needs test for null arguments (i.e., a MEF failure); `this.Host` and this.webBrowser` are assumed to never be null in this class.
Also, add the `[Import]` attribute to both ctor arguments (or remove them).